File: G-for-g-Practice/Indexes of Subarray Sum.java





class Solution {
    static ArrayList<Integer> subarraySum(int[] arr, int target) {
        // code here
        int left =0, currSum=0;
        ArrayList<Integer> result = new ArrayList<>();
        for (int right= 0; right<arr.length; right++){
            currSum+= arr[right];
            
            while(currSum > target && left <right){
                currSum-= arr[left];
                left++;
            }
            if (currSum==target){
                result.add(left+1);
                result.add(right+1);
                return result;
            }
        }
        result.add(-1);
        return result;
    }
}

File: G-for-g-Practice/Indexes of Subarray Sum.py
#User function Template for python3
class Solution:
    def subarraySum(self, arr, target):
        # code here
        left, curr_sum =0,0
        
        for right in range(len(arr)):
            curr_sum+= arr[right]
            
            while curr_sum >target and left<right:
                curr_sum -= arr[left]
                left+=1
            if curr_sum == target:
                return[left+1, right+1]
        return [-1]




